tests/target/issue-2506.rs

```diff
@@ -0,0 +1,22 @@
+#![feature(dyn_trait)]
+fn main() {
+    // checks rustfmt doesn't remove dyn
+    trait MyTrait {
+        fn method(&self) -> u64;
+    }
+    fn f1(a: Box<dyn MyTrait>) {}
+
+    // checks if line wrap works correctly
+    trait Very_______________________Long__________________Name____________________Trait
+         {
```

Comment:
The opening bracket should not be indented, but I guess this is a seperate issue. So this is fine as is.